app/models/collection.rb

```diff
@@ -4,5 +4,7 @@ class Collection < ApplicationRecord
 
   has_many :pockets, dependent: :destroy
 
+  delegate :organization, to: :route, allow_nil: true
```

Comment:
i'm not so sure if we should set `allow_nil` to true. Is that possible? I mean, is there a case in which the organization of a route is nil?

Reply:
This is because, when someone is trying to create a Collection with an invalid route_id, the Pockets associated to that collection are created first. They look for the organization, which is delegated to the collection, and the collection delegate this to the route that does not exist. Because of this, if **allow_nil is false**, an **Internal error server** is thrown instead of a **Bad request**.